src/worker/ghidra/sleighMishegos.cc

```diff
@@ -0,0 +1,261 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This class was copied from upstream
+ * https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Decompiler/src/decompile/cpp/sleigh.cc
+ *
+ * Modified by Eric Kilmer at Trail of Bits 2021
+ * Modified to better support mishegos single-shot disassembly by removing the
```

Comment:
Hmm, how come we query the same address multiple times? The way I'm doing it in Remill + `sleigh-lift` is by just appending to the `LoadImage` buffer before every read.

Does that not work for Mishegos use case?

Reply:
The other mishegos workers always use address 0, so I decided we should probably be consistent here as well.

Appending bytes to the address space in the mishegos use-case of fuzzing seems like it would cause unnecessary usage of memory when we really only care about a single instruction disassembly of a set of bytes.

If I didn't modify the default sleigh disassembler and just kept loading bytes into address `0x0`, then I would have hit the built-in (non-removable) caching mechanism that will return the first instruction that was decoded at address `0x0`, so loading new bytes into `0x0` wouldn't have any effect (until, of course, a certain number of other address disassemblies removed the `0x0` cache entry).

Reply:
Thanks for explaining. That makes sense.